ansible/cloud_providers/ec2_infrastructure_deployment.yml

```diff
@@ -95,6 +95,7 @@
     - name: Run infra-ec2-linux-set-hostname Role
       import_role:
         name: infra-ec2-linux-set-hostname
+      when: not install_cloud_provider  # ec2 cloud provider uses default hostnames
```

Comment:
You need to specify a default for this variable.

```yaml
install_cloud_provider | default(false)
```

This playbook is common to all configs, but you define `install_cloud_provider` only in ocp-workshop `env_vars.yml`.

Also, this variable only means something in the context of ocp-workshop config.
It should be done the other way around, in the ocp-workshop config:

- if install_cloud_provider is `true` then set `infra_set_hostname` to `false`. Or simply, in `env_vars.yml`:

```yaml
install_cloud_provider: false
infra_set_hostname: "{{ not install_cloud_provider }}"
```
And replace the test in `ec2_inrastructure_deployment.yml` with:

```yaml
when: infra_set_hostname | default(true)
```